datajoint/fetch.py

```diff
@@ -20,46 +25,57 @@ def prepare_attributes(relation, item):
         raise DataJointError("Index must be a slice, a tuple, a list, a string.")
     return item, attributes
 
-class FetchQuery:
+def copy_first(f):
```

Comment:
We thought that it would make more sense for each operation on the `fetch` object to return a new `fetch` object rather than modifying the object in place. Since `fetch` object in itself is very cheap, this shouldn't be a real over head. `copy_first` is a decorator that modifies the decorated method to work on and modify the copy of the object, supporting this idea.